internal/plugin/loader_test.go

```diff
@@ -268,3 +268,61 @@ func TestFindPlugins(t *testing.T) {
 		})
 	}
 }
+
+func TestLoadMetadataLegacy_CapitalNameField(t *testing.T) {
```

Comment:
Can we convert these into table drive tests? (stylistic, but also perhaps easier to maintain over time)